WEB/Src/WindowsServer/WindowsServer/AzureInstanceMetadataTelemetryModule.cs

```diff
@@ -15,17 +15,22 @@ public sealed class AzureInstanceMetadataTelemetryModule : ITelemetryModule
         private object lockObject = new object();
         private IHeartbeatPropertyManager heartbeatManager;
 
-        /// <summary>Gets a value indicating whether this module has been initialized.</summary>
-        internal bool IsInitialized { get; private set; } = false;
+        /// <summary>
+        /// Initializes a new instance of the<see cref="AzureInstanceMetadataTelemetryModule" /> class.
+        /// </summary>
+        public AzureInstanceMetadataTelemetryModule() : this(null)
+        {
+        }
+
+        internal AzureInstanceMetadataTelemetryModule(IHeartbeatPropertyManager heartbeatPropertyManager) => this.HeartbeatPropertyManager = heartbeatPropertyManager;
 
         /// <summary>
         /// Gets or sets an instance of IHeartbeatPropertyManager. 
         /// </summary>
         /// <remarks>
         /// This is expected to be an instance of <see cref="DiagnosticsTelemetryModule"/>.
-        /// Note that tests can also override the heartbeat manager.
         /// </remarks>
-        internal IHeartbeatPropertyManager HeartbeatPropertyManager
+        public IHeartbeatPropertyManager HeartbeatPropertyManager
```

Comment:
what are your thoughts on making this not Editorbrowseable? Its public, but we dont expect customers to every change it themselves right?

Reply:
I think that's an excellent idea.